src/server/index.js

```diff
@@ -1,155 +1,69 @@
-  next()
-})
-app.use(express.static(path.join(__dirname, '../../build/public'), {
-  maxAge: '1d'
-}))
-// host sw.js
-app.use(express.static(path.join(__dirname, '../../build/public/js')))
+function detectHttps() {
+  const certPath = resolve(process.cwd(), 'ssl/certificate.pem')
+  const keyPath = resolve(process.cwd(), 'ssl/private-key.pem')
+  if (!existsSync(certPath) || !existsSync(keyPath)) return {}
 
-// TODO reuse routes.js paths? depends on the server framework being used
-app.get('/', appShellHandler)
-app.get('/page/:pageNumber?', appShellHandler)
-app.get('/newest/:page?/:pageNumber?', appShellHandler)
-app.get('/newcomments/:page?/:pageNumber?', appShellHandler)
-app.get('/show/:page?/:pageNumber?', appShellHandler)
-app.get('/ask/:page?/:pageNumber?', appShellHandler)
-app.get('/jobs/:page?/:pageNumber?', appShellHandler)
-app.get('/app-shell', appShellHandler)
-app.get('/manifest.json', (request, response) => response.json(appManifest))
-app.get('/api/stories', (request, response) => {
-  const { filter, page } = request.query || {}
-
-  if (filter === 'comments') {
-    return RSSResponse(page, response)
-  }
-
-  return graphQLResponse(filter, page, response)
-})
-
-function sanitizeItemContent (items) {
-  return items.map(({content, ...rest}) => {
-    let cleanContent = sanitizeHtml(content)
-    return {
-      content: cleanContent,
-      ...rest
-    }
-  })
+  return { https: { key: readFileSync(keyPath), cert: readFileSync(certPath) }, http2: true }
 }
 
-function RSSResponse (page, response) {
-  const RSS_URL = 'https://hnrss.org/newcomments?count=100'
-  const parser = new Parser()
-  parser.parseURL(RSS_URL)
-    .then(feed => {
-      let startingItem = page ? (page - 1) * 30 : 0
-      const items = feed.items
-        .filter(item => item.title)
-        .slice(startingItem, startingItem + 30)
-      response.send(sanitizeItemContent(items))
-    })
-}
+function setupFetch(server) {
+  global.fetch = async function(url, params) {
+    // External call, just use node-fetch
+    if (!url.startsWith('/api')) return nodeFetch(url, params)
 
-function graphQLResponse (filter, page, response) {
-  const API_URL = 'https://www.graphqlhub.com/graphql/'
-  const offset = page > 1 ? (page - 1) * 30 : 0
-  let queryType
+    // Local fetching, serve with fastify.inject
+    const response = await server.inject({ method: 'GET', url, ...params })
 
-  switch (filter) {
-    case 'show':
-      queryType = 'showStories'
-      break
-    case 'ask':
-      queryType = 'askStories'
-      break
-    case 'jobs':
-      queryType = 'jobStories'
-      break
-    case 'rank':
-      queryType = 'newStories'
-      break
-    case 'new':
-      queryType = 'newStories'
-      break
-    case 'best':
-    default:
-      queryType = 'topStories'
+    response.status = response.statusCode
+    response.statusText = response.statusMessage
+    return new nodeFetch.Response(response.payload, response)
   }
+}
```

Comment:
Yup, it will be easy once converted to a plugin.

Reply:
It could already be passed through in `server.register(require('./api'))`.

Reply:
If I see this correct it is added as global because of how the client code uses it. I did not check closely how it would be possible to pass it through to that code, so I only checked the other call site and that is in the `api` server code. Instead of using it as a global inside of that file, it could be passed to the api by e.g. calling it as `server.register((fastify, opts) => require('./api')(fastify, opts, fetch))`.